apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/ApolloClient.kt

```diff
@@ -50,4 +55,100 @@ class ApolloClient(
         executionContext = executionContext + coroutineDispatcherContext
     )
   }
+
+  fun newBuilder(): Builder {
+    return DefaultBuilder(scalarTypeAdapters)
+        .networkTransport(networkTransport)
+        .subscriptionNetworkTransport(subscriptionNetworkTransport)
+        .interceptors(interceptors)
+        .executionContext(executionContext)
+  }
+
+  class DefaultBuilder(override val scalarTypeAdapters: ScalarTypeAdapters = ScalarTypeAdapters.DEFAULT) : Builder()
```

Comment:
is there a reason to have this `DefaultBuilder ` class, can't `Builder` be concrete class with `scalarTypeAdapters: ScalarTypeAdapters = ScalarTypeAdapters.DEFAULT`

Reply:
I was contemplating the idea of auto generating Builders with pre-built ScalarTypeAdapters like `ApolloClient.Companion.GithubBuilder()`. But that makes the generated code depend on the runtime which isn't great. I'll revert.